src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -707,4 +722,29 @@ public AllMapData getAll(UUID networkUuid, String variantId, List<String> substa
             return res.stream().collect(Collectors.toList());
         }
     }
+
+    public List<VoltageLevelMapData> getVoltageLevels(UUID networkUuid, String variantId, List<String> substationsId) {
+        Network network = getNetwork(networkUuid, substationsId == null ? PreloadingStrategy.COLLECTION : PreloadingStrategy.NONE, variantId);
+        if (substationsId == null) {
+            return network.getVoltageLevelStream()
+                .map(NetworkMapService::toMapData).collect(Collectors.toList());
+        } else {
+            Set<VoltageLevelMapData> res = new LinkedHashSet<>();
+            substationsId.stream().forEach(id ->
+                network.getSubstation(id).getVoltageLevelStream().forEach(v -> res.add(toMapData(v))));
```

Comment:
Check is not done in all other already existing methods ...
Substation with this id should exist in the network, because it's is generated by the network-modification-server ...
But, you're right, it would be safer to add a check in all the methods ...